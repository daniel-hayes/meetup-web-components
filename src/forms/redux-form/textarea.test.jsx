import React from 'react';
import { shallow } from 'enzyme';
import ReduxFormTextarea from './Textarea';

describe('redux-form Textarea', () => {
	const formAttrs = {
		input: {
			id: 'heroField',
			label: 'Super Hero',
			name: 'superhero',
			value: 'Wonder Woman and Robin',
			autosize: true,
			maxLength: 20,
			required: true,
		},
		meta: {
			error: 'Did you mean Batman and Robin?',
		},
	};

	it('renders a Textarea component with expected attributes from mock data', () => {
		const component = shallow(<ReduxFormTextarea {...formAttrs} />);
		expect(component).toMatchSnapshot();
	});
});
