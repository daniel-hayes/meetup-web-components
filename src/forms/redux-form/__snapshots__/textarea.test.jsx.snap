// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux-form Textarea renders a Textarea component with expected attributes from mock data 1`] = `
<WithErrorList(Textarea)
  autosize={true}
  error={null}
  id="heroField"
  label="Super Hero"
  maxLength={20}
  name="superhero"
  required={true}
  value="Wonder Woman and Robin"
/>
`;
